### ISSUE TITLE
Fetch after Select applies to wrong entity

### ISSUE THREAD

**Issue by bahusoid** [nhibernate/nhibernate-core#3356]
See `FetchAfterSelect` test case added with GH3352 test case.

Query source is wrongly detected and fetch is applied to main entity instead of selected.

**Comment by bahusoid** [comment]
Yeah. Later wrong HQL node is used for applying fetch. I already have a WIP fix for this issue.

**Comment by hazzik** [comment]
@bahusoid should this be closed?

**Comment by bahusoid** [comment]
> @bahusoid should this be closed?

Nope. Autoclosed by PR merge.
